### PR TITLE
Add support for setting Sensors on Android Emulators

### PR DESCRIPTION
**Pull request by vrunoa**
First step to adding support for setting sensor values on Android Emulators.

### CODE REVIEW

**Comment by jlipps** [comment]
nice! cc @isonic1

File: lib/tools/adb-emu-commands.js [hunk 1]
```diff
@@ -97,6 +106,22 @@ emuMethods.powerAC = async function powerAC (state = 'on') {
...
 };
 
+/**
+ * Emulate sensors values on the connected emulator.
+ *
+ * @param {string} sensor - Sensor type: light, pressure, humidity, etc.
+ * @param {float} value  - Number to set as the sensor value.
+ */
+emuMethods.sensorSet = async function sensorSet (sensor, value) {
+  if (_.values(emuMethods.SENSORS).indexOf(sensor) === -1) {
```
Comment (mykola-mokhnach) [inline lib/tools/adb-emu-commands.js]: use `includes`

File: lib/tools/adb-emu-commands.js [hunk 2]
```diff
@@ -97,6 +106,22 @@ emuMethods.powerAC = async function powerAC (state = 'on') {
...
+    throw new Error(`Unsupported sensor sent '${sensor}'. Supported values: ${_.values(emuMethods.SENSORS)}]`);
+  }
+  if (!_.isNumber(value)) {
```
Comment (mykola-mokhnach) [inline lib/tools/adb-emu-commands.js]: use isInteger

Comment (mykola-mokhnach) [inline lib/tools/adb-emu-commands.js]: apply parseFloat  to value and then !isNaN

File: lib/tools/adb-emu-commands.js [hunk 3]
```diff
@@ -97,6 +106,22 @@ emuMethods.powerAC = async function powerAC (state = 'on') {
   await this.adbExecEmu(['power', 'ac', state]);
 };
 
+/**
+ * Emulate sensors values on the connected emulator.
+ *
+ * @param {string} sensor - Sensor type: light, pressure, humidity, etc.
+ * @param {float} value  - Number to set as the sensor value.
```
Comment (mykola-mokhnach) [inline lib/tools/adb-emu-commands.js]: float|string

File: lib/tools/adb-emu-commands.js [hunk 4]
```diff
@@ -97,6 +106,22 @@ emuMethods.powerAC = async function powerAC (state = 'on') {
   await this.adbExecEmu(['power', 'ac', state]);
 };
 
+/**
+ * Emulate sensors values on the connected emulator.
+ *
+ * @param {string} sensor - Sensor type: light, pressure, humidity, etc.
```
Comment (mykola-mokhnach) [inline lib/tools/adb-emu-commands.js]: I'd rather say it should be one of SENSORS items

File: lib/tools/adb-emu-commands.js [hunk 5]
```diff
@@ -97,6 +106,22 @@ emuMethods.powerAC = async function powerAC (state = 'on') {
...
+ * @param {float} value  - Number to set as the sensor value.
+ */
```
Comment (mykola-mokhnach) [inline lib/tools/adb-emu-commands.js]: missing `throws` description

**Comment by isonic1** [comment]
Good stuff @vruno @mykola-mokhnach  @jlipps. Looking forward to trying it out. I put the link below to the integration I made for this in case it's helpful for anyone. 

https://github.com/isonic1/NTD/blob/master/android_sensor.rb

**Comment by vrunoa** [comment]
@isonic1 cool. The idea is to start adding this methods as commands on each of the clients, so everyone can use them, not just the people that know the adb/telnet. And adding an endpoint for each one of them. 
Moving forward, more and more platforms are supporting this sensors and IMO we should have a well defined methods/endpoints, Im going to try to bring up a proposal that @jlipps encourage me to do a few years ago. https://github.com/SeleniumHQ/mobile-spec/pull/6. Im going to start adding the new endpoints these days. Hope to hear feedback 😄

File: lib/tools/adb-emu-commands.js [hunk 6]
```diff
@@ -97,6 +106,24 @@ emuMethods.powerAC = async function powerAC (state = 'on') {
   await this.adbExecEmu(['power', 'ac', state]);
 };
 
+/**
+ * Emulate sensors values on the connected emulator.
+ *
+ * @param {string} sensor - Sensor type declared in SENSORS items.
+ * @param {float|string} value  - Number to set as the sensor value.
+ * @throws {error} - If sensor type or sebsor value is not defined
```
Comment (mykola-mokhnach) [inline lib/tools/adb-emu-commands.js]: Error

Comment (mykola-mokhnach) [reply lib/tools/adb-emu-commands.js]: seBsor

File: lib/tools/adb-emu-commands.js [hunk 7]
```diff
@@ -97,6 +106,24 @@ emuMethods.powerAC = async function powerAC (state = 'on') {
   await this.adbExecEmu(['power', 'ac', state]);
 };
 
+/**
+ * Emulate sensors values on the connected emulator.
+ *
+ * @param {string} sensor - Sensor type declared in SENSORS items.
+ * @param {float|string} value  - Number to set as the sensor value.
```
Comment (mykola-mokhnach) [inline lib/tools/adb-emu-commands.js]: javascript has only `number` type. there's no float

**Pull request merged**